arduino-core/test/processing/app/helpers/FileUtilsTest.java

```diff
@@ -0,0 +1,292 @@
+package processing.app.helpers;
+
+import static org.mockito.AdditionalMatchers.or;
+import static org.mockito.Matchers.*;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+
+import java.io.File;
+
+import java.nio.file.Path;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Random;
+
+@RunWith(PowerMockRunner.class)
+public class FileUtilsTest {
+
+	@Rule
+	public final ExpectedException thrown = ExpectedException.none();
+
+	@Rule public final Timeout globalTimeout = new Timeout(10000);
+
+  @Test
+  public void testIsSubDirectory1() throws Exception {
+    final File base = PowerMockito.mock(File.class);
+    final File file = PowerMockito.mock(File.class);
+    final File child = PowerMockito.mock(File.class);
+    final File file1 = PowerMockito.mock(File.class);
+
+    PowerMockito.when(file.equals(or(Matchers.isA(Object.class),
+      Matchers.isNull(Object.class)))).thenReturn(false);
```

Comment:
Something funny going on here.... I am quite sure you don't need this if the files are different

Reply:
I left this comment as a reminder for me -_-

Reply:
Do this instead

`  @Test
  public void testIsSubDirectory2() throws Exception {
    final File base = PowerMockito.mock(File.class);
    final File file = PowerMockito.mock(File.class);
    PowerMockito.when(base.getCanonicalFile()).thenReturn(file);

    Assert.assertTrue(FileUtils.isSubDirectory(base, base));
  }`